src/lib.mo

```diff
@@ -112,7 +112,7 @@ module {
     var cnt = count;
     while (cnt > 0) {
       let db_size = data_block_size(vec.i_block);
-      if (vec.i_element == 0 and db_size <= cnt) {
+      if (vec.i_element == 0 and db_size <= cnt and vec.data_blocks[vec.i_block].size() == 0) {
```

Comment:
I don't understand this change. I thought the old code was intentionally as it was. When the whole block needs to be filled then we do it by discarding the old block and creating a new one with Array.init. That was the idea (an optimization).